### PR TITLE
Add sci-geosciences/geographiclib

### PR DESCRIPTION
**Pull request by gemarcano**
Adds geographiclib as a new package for the lunar ROS distribution. One of the requirements for mavros. See issue #229.

### CODE REVIEW

**Comment by allenh1** [comment]
This looks good, but I'm not so sure it should be a ros package -- should this be under `sci-lib` and then listed as a dependency?

**Comment by gemarcano** [comment]
For bloom and superflore to pick it up for dependency stuff, it doesn't need to be a ROS package? Then sure, I can move it. Should I remove the ros-cmake eclass usage then?

**Comment by gemarcano** [comment]
Oh, and does this ebuild/package still need a package.xml?

**Comment by allenh1** [comment]
> For bloom and superflore to pick it up for dependency stuff, it doesn't need to be a ROS package?

That is correct, yes. It just needs to have its rosdep key listed in ros/rosdistro.

> Should I remove the ros-cmake eclass usage then?

Ya, just plain old `cmake` is fine.

> Oh, and does this ebuild/package still need a package.xml?

It shouldn't. Just a normal, ros-agnostic ebuild is all this ebuild needs to be.

Thanks for the work you've done! Excited to cross another dependency off the list!

**Comment by gemarcano** [comment]
Hm, since we don't need the ROS specific commits anymore, I'm going to do a rebase and get rid of them. I'll then force push the changes up.

File: sci-libs/geographiclib/geographiclib-1.49.ebuild [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit versionator
+inherit cmake-utils
+
+PYTHON_COMPAT=( python{2_7,3_4} )
```
Comment (allenh1) [inline sci-libs/geographiclib/geographiclib-1.49.ebuild]: is `3_5` possible as well?

File: sci-libs/geographiclib/geographiclib-1.49.ebuild [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+MY_PV="$(replace_version_separator 2 '-')"
+MY_P="${MY_PN}-${MY_PV}"
+
+DESCRIPTION=""
+HOMEPAGE="https://sourceforge.net/projects/geographiclib/"
+SRC_URI="https://sourceforge.net/projects/geographiclib/files/distrib/${MY_P}.tar.gz/download -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
```
Comment (allenh1) [inline sci-libs/geographiclib/geographiclib-1.49.ebuild]: Could you also add `~arm ~arm64 ~x86`?

File: sci-libs/geographiclib/geographiclib-1.49.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+inherit python-r1
+
+MY_PN="GeographicLib"
+MY_PV="$(replace_version_separator 2 '-')"
+MY_P="${MY_PN}-${MY_PV}"
+
+DESCRIPTION=""
+HOMEPAGE="https://sourceforge.net/projects/geographiclib/"
+SRC_URI="https://sourceforge.net/projects/geographiclib/files/distrib/${MY_P}.tar.gz/download -> ${P}.tar.gz"
```
Comment (allenh1) [inline sci-libs/geographiclib/geographiclib-1.49.ebuild]: Nit: `https://sourceforge.net/projects/${PN}/files/distrib/${MY_P}.tar.gz/download -> ${P}.tar.gz`

File: sci-libs/geographiclib/geographiclib-1.49.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+PYTHON_COMPAT=( python{2_7,3_4} )
+
+inherit python-r1
+
+MY_PN="GeographicLib"
+MY_PV="$(replace_version_separator 2 '-')"
+MY_P="${MY_PN}-${MY_PV}"
+
+DESCRIPTION=""
```
Comment (allenh1) [inline sci-libs/geographiclib/geographiclib-1.49.ebuild]: Could you please include a brief description here?

File: sci-libs/geographiclib/geographiclib-1.49.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+src_configure() {
+	local mycmakeargs=(
+		-DGEOGRAPHICLIB_DOCUMENTATION=$(usex doc ON OFF)
+		-DGEOGRAPHICLIB_LIB_TYPE=$(usex static-libs BOTH SHARED)
+	)
+
+	cmake-utils_src_configure
+}
+
```
Comment (allenh1) [inline sci-libs/geographiclib/geographiclib-1.49.ebuild]: I think this is an extra newline, but it's kinda hard to tell. If so, please remove it.

File: sci-libs/geographiclib/geographiclib-1.49.ebuild (hunk 1 above)
Comment (gemarcano) [reply sci-libs/geographiclib/geographiclib-1.49.ebuild]: Sure, there should be no reason not to, other than I haven't really tested it. And also that this package installs the python scripts into a non-versioned `site-packages/python` path.

File: sci-libs/geographiclib/geographiclib-1.49.ebuild (hunk 2 above)
Comment (gemarcano) [reply sci-libs/geographiclib/geographiclib-1.49.ebuild]: Sure, but I haven't tested this on those systems.

**Comment by gemarcano** [comment]
Alright, that should do it. I remembered about `repoman` and that reminded me about the metadata.xml file. I think this should do it.

**Comment by gemarcano** [comment]
Hm, I decided to scour through Gentoo's package category list, and found a possible better match for the category. Should we move this package to `sci-geosciences`?

**Pull request merged**